Add support for Docker Compose development environment

hectcastro opened the pull request:
Include the Docker configuration files necessary to support a PostgreSQL backed instance of the application running on Ruby 2.3.

Fixes https://github.com/andrew/octobox/issues/12

WillAbides commented on docker-compose.yml:
```diff
@@ -0,0 +1,25 @@
+      context: ./
+      dockerfile: Dockerfile
+    ports:
+      - 3000:3000
```
It would be nice to see something like `- ${GITHUB_INBOX_PORT:-3000}:3000` here, but that requires using version 2.1 of the docker-compose.yml format, which in turn requires docker-compose >= 1.9.  I'm not sure if enough people are on 1.9 yet.

hectcastro replied:
I'm leaning toward leaving it as is to that things will work for a wider audience, but maybe @andrew can weigh in.

(I did not know about Docker Compose 1.9, or the update to this behavior in the Docker Compose YAML format, so thanks for the heads up about it.)

WillAbides replied:
I think you're right.  If 1.9 was a few months older, it would be worth doing.  Maybe in April...

andrew commented:
I've just [renamed](https://github.com/andrew/octobox/pull/38) the project to Octobox, can you update the database and ENV config names?

calavera commented on Dockerfile:
```diff
@@ -0,0 +1,6 @@
+
+RUN \
+    apt-get update \
+    && apt-get install -y --no-install-recommends nodejs \
```
you might want to use `apt-get clean` here to remove archives from the apt cache:

```
clean
    clean clears out the local repository of retrieved package files. It removes everything but the lock file from /var/cache/apt/archives/ and /var/cache/apt/archives/partial/
````

hectcastro replied:
Addressed in 47a499af66ac09238544a050d9cc2e6c25e65d3a.

hectcastro commented:
I think I've addressed all of the comments.

Regarding the `docker-compose` commands for `database` and `app` in the `README`, they're split to avoid the Rails application from attempting to connect to the PostgreSQL database before it is completely initialized.

andrew commented:
Thanks @hectcastro 🎉

bfirsh commented on README.md:
````diff
@@ -87,7 +87,23 @@ Finally you can boot the rails app:
 ```
+#### Docker Compose
 
+If you're familiar with [Docker](https://docs.docker.com/engine/) and [Docker Compose](https://docs.docker.com/compose/), the included `docker-compose.yml` configuration allows you to spin up the application locally.
````
Probably best to link to: https://docs.docker.com/installation/

Docker is actually of more value if you're not familiar with it. My normal instructions here are:

Step 1: [install Docker for Mac/Windows.](https://docs.docker.com/installation/)
Step 2: run `docker-compose up`

Boom.

andrew replied:
Thanks for the tips @bfirsh, I don't suppose fancy sending the changes as a pr?

bfirsh replied:
👍 will have a look later

bfirsh commented on README.md:
````diff
@@ -87,7 +87,23 @@ Finally you can boot the rails app:
+docker-compose up database 
+```
+
+Once the PostgreSQL initialization process is complete, launch the application using another terminal session:
````
You can use netcat to avoid this dance: http://stackoverflow.com/questions/31746182/docker-compose-wait-for-container-x-before-starting-y
